### DISCUSSION TITLE
[@types/bootstrap] collapse.d.ts:82:36 - error TS2304: Cannot find name 'JQuery'

### DISCUSSION THREAD

**Discussion by marcioggs** [DefinitelyTyped/DefinitelyTyped#56194] [Issues with a @types package]
The following error is occurring when compiling an Angular 12 application:

```
[INFO] > ng build --progress false
[INFO]
[INFO]
[INFO] Error: node_modules/@types/bootstrap/js/dist/collapse.d.ts:82:36 - error TS2304: Cannot find name 'JQuery'.
[INFO]
[INFO] 82         parent: string | Element | JQuery;
```

I could make it work by adding this import to `collapse.d.ts`:
```
import * as JQuery from 'jquery';
```
Do I have to make a similar external configuration to make it work or is it really an issue with the @types/bootstrap module?

package.json:
```
    "@types/bootstrap": "^5.1.6",
```

package-lock.json:
```
    "@types/bootstrap": {
      "version": "5.1.6",
      "resolved": "https://registry.npmjs.org/@types/bootstrap/-/bootstrap-5.1.6.tgz",
      "integrity": "sha512-3L6IvOCKyoVd3e4bgQTH7VBPbuYEOG8IQbRcuZ0AbjfwPdRX+kVf5L/7mVt1EVM+D/BVw4+71rtp7Z8yYROlpQ==",
      "dev": true,
      "requires": {
        "@popperjs/core": "^2.9.2",
        "@types/jquery": "*"
      }
    },

   ...

       "@types/jquery": {
      "version": "3.5.6",
      "resolved": "https://registry.npmjs.org/@types/jquery/-/jquery-3.5.6.tgz",
      "integrity": "sha512-SmgCQRzGPId4MZQKDj9Hqc6kSXFNWZFHpELkyK8AQhf8Zr6HKfCzFv9ZC1Fv3FyQttJZOlap3qYb12h61iZAIg==",
      "dev": true,
      "requires": {
        "@types/sizzle": "*"
      }
    },
```

[a comment by an automated account is not shown]

**Reply by peterblazejewicz** [reply]
There is no special setup involved. one should just import (ESM) symbol and use it.
I've tried latest ng version to be sure there is no deps issue, with barebone collapse code:

```
❯❯ test-collapse git:(master)  22:21 npx ng build --progress false

Initial Chunk Files               | Names         |      Size
main.1acca5442670dfc0d379.js      | main          | 264.24 kB
styles.19c79e5e360791e1c3b9.css   | styles        | 156.57 kB
polyfills.d8c967ccda5d64970322.js | polyfills     |  36.21 kB
runtime.40b3fc8bebd8f3034578.js   | runtime       |   1.20 kB

                                  | Initial Total | 458.21 kB
```

and on sandbox:
https://codesandbox.io/s/bootstrap-collapse-angular-1egmc

One is supposed to just add 'bootstrap' and either wait for automatic type acquisition to kick in or add @types/bootstrap. Nothing more is required (like changes in the tsconfig)

**Reply by peterblazejewicz** [reply]
and there is no dependency on jQuery to be installed. @types/jquery are added via types resolution, but there is no requirement to add native jquery package in BS5

**Comment by marcioggs** [comment]
Possibly this only occurs when trying to optimize following the Lean JavaScript instructions:
https://getbootstrap.com/docs/5.1/customize/optimize/#lean-javascript

**Comment by peterblazejewicz** [comment]
this one also should work, just minimizing bundle size:

![image](https://user-images.githubusercontent.com/14539/136835795-81297b04-09d3-4fb2-976f-52e9be9bc6d6.png)
![image](https://user-images.githubusercontent.com/14539/136836012-7617e465-83d6-4fd6-b9cd-d25d95bffbc9.png)

so I'm just wondering what is the setup you're using. Can you make sure to remove any staled cache module for @types/bootstrap?
